pkg/skaffold/errors/err_map.go

```diff
@@ -87,3 +87,24 @@ var knownBuildProblems = []problem{
 		},
 	},
 }
+
+// Deploy errors in deployment phase
+var knownDeployProblems = []problem{
+	{
+		regexp:  re("(?i).*unable to connect.*: Get (.*)"),
+		errCode: proto.StatusCode_DEPLOY_CLUSTER_CONNECTION_ERR,
+		description: func(err error) string {
+			matchExp := re("(?i).*unable to connect.*Get (.*)")
+			if match := matchExp.FindStringSubmatch(fmt.Sprintf("%s", err)); len(match) >= 2 {
+				return fmt.Sprintf("Deploy Failed. Could not connect to cluster %s due to %s", skaffoldOpts.KubeContext, match[1])
+			}
+			return fmt.Sprintf("Deploy Failed. Could not connect to Kubernetes cluster.")
+		},
+		suggestion: func(opts config.SkaffoldOptions) []*proto.Suggestion {
+			return []*proto.Suggestion{{
```

Comment:
use `util.IsMinikube` to suggest different actions like i mentioned in https://github.com/GoogleContainerTools/skaffold/pull/4933#issuecomment-716888249